File: GoWebAPI/todo.go
package main

import "time"

type Todo struct {
	Id         int        `json:"id"`
	Title      *string    `json:"title,omitempty"`
	DueBy      *time.Time `json:"dueBy,omitempty"`
	IsComplete bool       `json:"isComplete"`
}

func GetToDos() []Todo {

	now := time.Now()
	sampleTodos := []Todo{
		{Id: 1, Title: strPtr("Set up a new Go 1.23 project"), DueBy: datePtr(now)},
		{Id: 2, Title: strPtr("Install necessary NuGet packages"), DueBy: datePtr(now.AddDate(0, 0, 1))},
		{Id: 3, Title: strPtr("Create solution file"), DueBy: datePtr(now.AddDate(0, 0, 2))},
		{Id: 4, Title: strPtr("Add project references for database"), DueBy: datePtr(now.AddDate(0, 0, 3))},
		{Id: 5, Title: strPtr("Configure logging services"), DueBy: datePtr(now.AddDate(0, 0, 4))},
		{Id: 6, Title: strPtr("Set up dependency injection"), DueBy: datePtr(now.AddDate(0, 0, 5))},
		{Id: 7, Title: strPtr("Implement basic user authentication"), DueBy: datePtr(now.AddDate(0, 0, 6))},
		{Id: 8, Title: strPtr("Add validation for input fields"), DueBy: datePtr(now.AddDate(0, 0, 7))},
		{Id: 9, Title: strPtr("Create unit tests for validation"), DueBy: datePtr(now.AddDate(0, 0, 8))},
		{Id: 10, Title: strPtr("Add middleware for error handling"), DueBy: datePtr(now.AddDate(0, 0, 9))},
		{Id: 11, Title: strPtr("Design the database schema"), DueBy: datePtr(now.AddDate(0, 0, 10))},
		{Id: 12, Title: strPtr("Create the database context"), DueBy: datePtr(now.AddDate(0, 0, 11))},
		{Id: 13, Title: strPtr("Implement repository pattern"), DueBy: datePtr(now.AddDate(0, 0, 12))},
		{Id: 14, Title: strPtr("Add logging to repository methods"), DueBy: datePtr(now.AddDate(0, 0, 13))},
		{Id: 15, Title: strPtr("Create models for API response"), DueBy: datePtr(now.AddDate(0, 0, 14))},
		{Id: 16, Title: strPtr("Set up Swagger for API documentation"), DueBy: datePtr(now.AddDate(0, 0, 15))},
		{Id: 17, Title: strPtr("Create sample controller"), DueBy: datePtr(now.AddDate(0, 0, 16))},
		{Id: 18, Title: strPtr("Add actions for CRUD operations in controller"), DueBy: datePtr(now.AddDate(0, 0, 17))},
		{Id: 19, Title: strPtr("Implement data seeding for testing"), DueBy: datePtr(now.AddDate(0, 0, 18))},
		{Id: 20, Title: strPtr("Create custom exception class"), DueBy: datePtr(now.AddDate(0, 0, 19))},
		{Id: 21, Title: strPtr("Set up JWT authentication"), DueBy: datePtr(now.AddDate(0, 0, 20))},
		{Id: 22, Title: strPtr("Test JWT authentication middleware"), DueBy: datePtr(now.AddDate(0, 0, 21))},
		{Id: 23, Title: strPtr("Add user roles and permissions"), DueBy: datePtr(now.AddDate(0, 0, 22))},
		{Id: 24, Title: strPtr("Create the user registration API"), DueBy: datePtr(now.AddDate(0, 0, 23))},
		{Id: 25, Title: strPtr("Implement password hashing"), DueBy: datePtr(now.AddDate(0, 0, 24))},
		{Id: 26, Title: strPtr("Create user login API"), DueBy: datePtr(now.AddDate(0, 0, 25))},
		{Id: 27, Title: strPtr("Test login functionality"), DueBy: datePtr(now.AddDate(0, 0, 26))},
		{Id: 28, Title: strPtr("Create an API endpoint for fetching user profile"), DueBy: datePtr(now.AddDate(0, 0, 27))},
		{Id: 29, Title: strPtr("Set up auto-mapping between DTOs and models"), DueBy: datePtr(now.AddDate(0, 0, 28))},
		{Id: 30, Title: strPtr("Add logging for user actions"), DueBy: datePtr(now.AddDate(0, 0, 29))},
		{Id: 31, Title: strPtr("Implement pagination for list views"), DueBy: datePtr(now.AddDate(0, 0, 30))},
		{Id: 32, Title: strPtr("Create model for user preferences"), DueBy: datePtr(now.AddDate(0, 0, 31))},
		{Id: 33, Title: strPtr("Add validation to user preference model"), DueBy: datePtr(now.AddDate(0, 0, 32))},
		{Id: 34, Title: strPtr("Create API endpoint for updating user preferences"), DueBy: datePtr(now.AddDate(0, 0, 33))},
		{Id: 35, Title: strPtr("Test updating user preferences endpoint"), DueBy: datePtr(now.AddDate(0, 0, 34))},
		{Id: 36, Title: strPtr("Add caching to frequently accessed endpoints"), DueBy: datePtr(now.AddDate(0, 0, 35))},
		{Id: 37, Title: strPtr("Set up caching policies"), DueBy: datePtr(now.AddDate(0, 0, 36))},
		{Id: 38, Title: strPtr("Create custom middleware for rate limiting"), DueBy: datePtr(now.AddDate(0, 0, 37))},
		{Id: 39, Title: strPtr("Write integration tests for API endpoints"), DueBy: datePtr(now.AddDate(0, 0, 38))},
		{Id: 40, Title: strPtr("Implement email sending functionality"), DueBy: datePtr(now.AddDate(0, 0, 39))},
		{Id: 41, Title: strPtr("Create templates for email notifications"), DueBy: datePtr(now.AddDate(0, 0, 40))},
		{Id: 42, Title: strPtr("Integrate with an SMTP server"), DueBy: datePtr(now.AddDate(0, 0, 41))},
		{Id: 43, Title: strPtr("Set up environment variables for email credentials"), DueBy: datePtr(now.AddDate(0, 0, 42))},
		{Id: 44, Title: strPtr("Create password reset API"), DueBy: datePtr(now.AddDate(0, 0, 43))},
		{Id: 45, Title: strPtr("Test password reset functionality"), DueBy: datePtr(now.AddDate(0, 0, 44))},
		{Id: 46, Title: strPtr("Implement 2FA for user authentication"), DueBy: datePtr(now.AddDate(0, 0, 45))},
		{Id: 47, Title: strPtr("Set up Azure DevOps pipeline for CI/CD"), DueBy: datePtr(now.AddDate(0, 0, 46))},
		{Id: 48, Title: strPtr("Configure build and release pipelines"), DueBy: datePtr(now.AddDate(0, 0, 47))},
		{Id: 49, Title: strPtr("Add deployment scripts for staging environment"), DueBy: datePtr(now.AddDate(0, 0, 48))},
		{Id: 50, Title: strPtr("Configure database migration in CI/CD pipeline"), DueBy: datePtr(now.AddDate(0, 0, 49))},
	}

	return sampleTodos
}

func strPtr(s string) *string {
	return &s
}
func datePtr(t time.Time) *time.Time {
	return &t
}

File: DenoWebAPI/todo.ts
interface Todo {
  id: number;
  title: string;
  dueBy: string; 
 isComplete:boolean;
}
  const sampleTodos: Todo[] = [
    { id: 1, title: "Set up a new Deno 2 project", dueBy: new Date().toISOString().split('T')[0] , isComplete: false},
    { id: 2, title: "Install necessary NuGet packages", dueBy: new Date(Date.now() + 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 3, title: "Create solution file", dueBy: new Date(Date.now() + 2 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 4, title: "Add project references for database", dueBy: new Date(Date.now() + 3 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 5, title: "Configure logging services", dueBy: new Date(Date.now() + 4 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 6, title: "Set up dependency injection", dueBy: new Date(Date.now() + 5 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 7, title: "Implement basic user authentication", dueBy: new Date(Date.now() + 6 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 8, title: "Add validation for input fields", dueBy: new Date(Date.now() + 7 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 9, title: "Create unit tests for validation", dueBy: new Date(Date.now() + 8 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 10, title: "Add middleware for error handling", dueBy: new Date(Date.now() + 9 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 11, title: "Design the database schema", dueBy: new Date(Date.now() + 10 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 12, title: "Create the database context", dueBy: new Date(Date.now() + 11 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 13, title: "Implement repository pattern", dueBy: new Date(Date.now() + 12 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 14, title: "Add logging to repository methods", dueBy: new Date(Date.now() + 13 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 15, title: "Create models for API response", dueBy: new Date(Date.now() + 14 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 16, title: "Set up Swagger for API documentation", dueBy: new Date(Date.now() + 15 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 17, title: "Create sample controller", dueBy: new Date(Date.now() + 16 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 18, title: "Add actions for CRUD operations in controller", dueBy: new Date(Date.now() + 17 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 19, title: "Implement data seeding for testing", dueBy: new Date(Date.now() + 18 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 20, title: "Create custom exception class", dueBy: new Date(Date.now() + 19 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 21, title: "Set up JWT authentication", dueBy: new Date(Date.now() + 20 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 22, title: "Test JWT authentication middleware", dueBy: new Date(Date.now() + 21 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 23, title: "Add user roles and permissions", dueBy: new Date(Date.now() + 22 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 24, title: "Create the user registration API", dueBy: new Date(Date.now() + 23 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 25, title: "Implement password hashing", dueBy: new Date(Date.now() + 24 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 26, title: "Create user login API", dueBy: new Date(Date.now() + 25 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 27, title: "Test login functionality", dueBy: new Date(Date.now() + 26 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 28, title: "Create an API endpoint for fetching user profile", dueBy: new Date(Date.now() + 27 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 29, title: "Set up auto-mapping between DTOs and models", dueBy: new Date(Date.now() + 28 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 30, title: "Add logging for user actions", dueBy: new Date(Date.now() + 29 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 31, title: "Implement pagination for list views", dueBy: new Date(Date.now() + 30 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 32, title: "Create model for user preferences", dueBy: new Date(Date.now() + 31 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 33, title: "Add validation to user preference model", dueBy: new Date(Date.now() + 32 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 34, title: "Create API endpoint for updating user preferences", dueBy: new Date(Date.now() + 33 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 35, title: "Test updating user preferences endpoint", dueBy: new Date(Date.now() + 34 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 36, title: "Add caching to frequently accessed endpoints", dueBy: new Date(Date.now() + 35 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 37, title: "Set up caching policies", dueBy: new Date(Date.now() + 36 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 38, title: "Create custom middleware for rate limiting", dueBy: new Date(Date.now() + 37 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 39, title: "Write integration tests for API endpoints", dueBy: new Date(Date.now() + 38 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 40, title: "Implement email sending functionality", dueBy: new Date(Date.now() + 39 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 41, title: "Create templates for email notifications", dueBy: new Date(Date.now() + 40 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 42, title: "Integrate with an SMTP server", dueBy: new Date(Date.now() + 41 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 43, title: "Set up environment variables for email credentials", dueBy: new Date(Date.now() + 42 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 44, title: "Create password reset API", dueBy: new Date(Date.now() + 43 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 45, title: "Test password reset functionality", dueBy: new Date(Date.now() + 44 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 46, title: "Implement 2FA for user authentication", dueBy: new Date(Date.now() + 45 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 47, title: "Set up Azure DevOps pipeline for CI/CD", dueBy: new Date(Date.now() + 46 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 48, title: "Configure build and release pipelines", dueBy: new Date(Date.now() + 47 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 49, title: "Add deployment scripts for staging environment", dueBy: new Date(Date.now() + 48 * 86400000).toISOString().split('T')[0] , isComplete: false},
    { id: 50, title: "Configure database migration in CI/CD pipeline", dueBy: new Date(Date.now() + 49 * 86400000).toISOString().split('T')[0] , isComplete: false}
  ];

  export { sampleTodos };  